scripts/anvil_tools/post_workspace_attributes.py

```diff
@@ -20,53 +21,147 @@ def get_access_token():
     return credentials.get_access_token().access_token
 
 
-def call_putLibraryMetadata_api(request, workspace_name, workspace_project):
+def write_output_report(workspace_status_dataframe):
+    """Report workspace set-up statuses and create output tsv file from provided dataframe."""
+
+    # create timestamp and use to label output file
+    timestamp = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    output_filename = f"{timestamp}_workspaces_published_status.tsv"
+    workspace_status_dataframe.to_csv(output_filename, sep="\t", index=False)
+
+    # count success and failed workspaces and report to stdout
+    successes = workspace_status_dataframe.workspace_setup_status.str.count("Success").sum()
+    fails = workspace_status_dataframe.workspace_setup_status.str.count("Failed").sum()
+    total = successes + fails
+    print(f"Number of workspaces passed set-up: {successes}/{total}")
+    print(f"Number of workspaces failed set-up: {fails}/{total}")
+    print(f"All workspace set-up (success or fail) details available in output file: {output_filename}")
+
+
+def publish_workspace(workspace_name, project="anvil-datastorage"):
```

Comment:
i'm confused about this vs `call_publish_library_workspace_api` in the other script? are these the same?